### ISSUE TITLE
list papers for author even if it's not the corresponding author

### ISSUE THREAD

**Issue by valearna** [WormBase/afp_authordash_webapp#1]
Retrieve papers linked to authors emails in postgres and not only afp_email table

**Comment by draciti** [comment]
@valearna, where can we test this?

**Comment by valearna** [comment]
I can't find any non-corresponding author with complete information in postgres. This means that we can only link the papers that we have processed to their corresponding author. For example, If I take the set of all papers that we emailed out with the new AFP pipeline, and then all the author ids in the pap_author table for these papers, I can't find any entries related to those authors in the two_email table. It sounds a bit strange to me, but maybe contact info for those papers have not been fully inserted yet? Any thoughts @draciti?

**Comment by draciti** [comment]
not sure about cecilia's curation pipeline.. does she capture e-mail only for corresponding authors?

**Comment by vanaukenk** [comment]
@draciti @valearna 
We should have emails for more than corresponding authors.  It may be, though, that until the author-person connections are verified, the information linking an author to a person and thus, their email, is not available.
In other words, this might just be a time delay.

**Comment by valearna** [comment]
From email conversation with Juancarlos, I found out that the query that I was using to retrieve the lists of papers for authors were incomplete. I'm working on them. This is a query that gives you the list of non-corresponding authors related to papers processed by the new afp pipeline:

select distinct two_email from two_email where joinkey in (select pap_author_possible from pap_author_possible where author_id in (select pap_author from pap_author where joinkey in (select joinkey from afp_version where afp_version = '2'))) AND two_email not in (select afp_email from afp_email where joinkey in (select joinkey from afp_version where afp_version = '2'));

I'll let you know when the interface is ready for testing.

**Comment by vanaukenk** [comment]
Thanks, @valearna

One thing we may want to check on though is if pap_author_possible can ultimately be different from the verified author, possibly in pap_author_verified (I'm not sure if that's the correct table).  If that changes, we might want to have a mechanism for updating.

**Comment by valearna** [comment]
yes, you are right @vanaukenk - what I'm doing now is the following: given an email address, I retrieve all the papers where that particular email address appears in afp_email (the emails that we send out to corresponding authors) + the papers where the email address is linked to a pap_author_possible entry. This means that authors see papers where they are corresponding authors OR listed as possible authors, verified or not. If we want to limit it to verified authors we have to wait for a curator to confirm the data. What do you prefer?

**Comment by vanaukenk** [comment]
@valearna
I think this probably depends on frequency of changes.  Let me ask Cecilia if she has an idea of that, or if the changes are tracked in any way to get a more precise number.  If it only happens a small percentage of the time, then I think it's worth going ahead with pap_author_possible.
What do you think?

**Comment by valearna** [comment]
I've just finished updating the api and http://textpressocentral.org:5003/ now lists papers for all authors. You can get the list of pap_author_possible linked to papers processed by the new afp who are not corresponding authors with the query above to test the interface.
